ClavaLaraApi/src-lara-clava/clava/clava/code/RemoveShadowing.js

```diff
@@ -0,0 +1,16 @@
+function RemoveShadowing($function) {
+  const usedNames = new Set();
+  let aliasIndex = 0;
+
+  for (const $param of $function.params) {
+    usedNames.add($param.name);
+  }
+
+  for (const $vardecl of $function.body.descendants("vardecl")) {
+    if (usedNames.has($vardecl.name)) {
+      const newName = `${$vardecl.name}_${aliasIndex++}`;
```

Comment:
Maybe we should consider centralizing generation of variable names, to ensure that those names are unique and do not conflict with existing names.